module/executiondatasync/execution_data/cache/cache.go

```diff
@@ -114,3 +114,24 @@ func (c *ExecutionDataCache) LookupID(blockID flow.Identifier) (flow.Identifier,
 
 	return result.ExecutionDataID, nil
 }
+
+func (c *ExecutionDataCache) ByExecutionResultID(ctx context.Context, resultID flow.Identifier) (*execution_data.BlockExecutionDataEntity, error) {
+	result, err := c.results.ByID(resultID)
+	if err != nil {
+		return nil, fmt.Errorf("failed to lookup execution result %s: %w", resultID, err)
+	}
+
+	if execData, ok := c.cache.Get(result.BlockID); ok {
+		return execData, nil
+	}
```

Comment:
We can't use the cache by block ID here since there could be execution forks. I think the thing to do is refactor the cache so that it stores by resultID, and update the other methods to lookup the sealed result for the block.